clang/lib/CIR/CodeGen/CIRRecordLayoutBuilder.cpp

```diff
@@ -580,7 +580,9 @@ CIRGenTypes::computeRecordLayout(const RecordDecl *D,
         builder.astRecordLayout.getSize()) {
       CIRRecordLowering baseBuilder(*this, D, /*Packed=*/builder.isPacked);
       auto baseIdentifier = getRecordTypeName(D, ".base");
-      *BaseTy = Builder.getStructTy(baseBuilder.fieldTypes, baseIdentifier,
+      auto recordKind = Builder.getRecordKind(D->getTagKind());
+      *BaseTy = Builder.getStructTy(recordKind, baseBuilder.fieldTypes,
```

Comment:
`getStructTy` should be able to call `getRecordKind(D->getTagKind())`, better move this there.